pootle/apps/pootle_statistics/migrations/0017_remove_duplicate_subs.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.5 on 2017-02-26 13:28
+from __future__ import unicode_literals
+
+import logging
+import time
+
+from django.db import migrations
+from django.db.models import Count
+
+
+logger = logging.getLogger(__name__)
+
+
+def remove_duplicate_subs(apps, schema_editor):
+    subs = apps.get_model("pootle_statistics.Submission").objects.all()
+    dupes = (
+        subs.order_by("creation_time", "unit_id", "type", "field")
+            .values("creation_time", "type", "field", "unit_id", "suggestion_id")
+            .distinct()
+            .annotate(Count("id"))
+            .filter(id__count__gt=1))
+    total = dupes.count()
+    start = time.time()
+    count = 0
+    for dupe in dupes.iterator():
+        count += 1
+        to_remove = list(
+            subs.filter(creation_time=dupe["creation_time"])
+                .filter(unit_id=dupe["unit_id"])
+                .filter(type=dupe["type"])
+                .filter(suggestion_id=dupe["suggestion_id"])
+                .filter(field=dupe["field"])
+                .values_list("id", flat=True))[1:]
+        subs.filter(id__in=to_remove).delete()
```

Comment:
its slower this way, but safer as otherwise the list of pks gets too long and "mysql goes away" and also we can swell memory

i can check the numbers tho - it might be safe to delete them together